radis/spectrum/spectrum.py

```diff
@@ -1632,6 +1632,24 @@ def rescale_mole_fraction(
             verbose=verbose,
         )
 
+    def _update_all_vars_from_abscoeff(self, new_abscoeff):
```

Comment:
similar to [_recompute_all_at_equilibrium](https://github.com/radis/radis/blob/develop/radis/spectrum/rescale.py#L828)

It is slightly more bulky / harder to read; but will make sure there aren't any unit mismatch, for instance.

Reply:
I'd keep your code and add it as a test, though. Recomputing these coeffiicents is a very sensitive part so it shouldn't fail, ever. It's already extensively tested in RADIS but there aren't too many such tests ever

Something like : 
```
# Recompute quantities
new_absorbance = new_abscoeff * self.conditions["path_length"]
new_transmittance_noslit = exp(-new_absorbance)
new_emissivity_noslit = (
  1 - new_transmittance_noslit
) 
new_radiance_noslit = calc_radiance(
  self.get_wavenumber(),  # TO-DO: Only true for Equilibirum!
  new_emissivity_noslit,
  self.conditions["Tgas"],
  unit=self.units["radiance_noslit"],
)

# Compare with arrays recomputed automatically 
assert (q.get("absorbance")[1] == new_absorbance).all()
assert (q.get("transmittance_noslit")[1] == new_transmittance_noslit).all()
etc.
```

Reply:
Good point! Did you include it already or to-do?